smarts/core/smarts.py

```diff
@@ -954,7 +954,7 @@ def providers(self) -> List[Provider]:
         # TODO: It's inconsistent that pybullet is not here
         return self._providers
 
-    def get_provider_by_type(self, requested_type) -> Provider:
+    def get_provider_by_type(self, requested_type):
```

Comment:
I would suggest this instead be:  `-> Optional[Provider]`

Reply:
I have the same opinion as @sah-huawei